chunky/src/java/se/llbit/chunky/main/Chunky.java

```diff
@@ -405,23 +395,35 @@ public SceneFactory getSceneFactory() {
   }
 
   @PluginApi
-  public void setPreviewRayTracerFactory(RayTracerFactory previewRayTracerFactory) {
-    this.previewRayTracerFactory = previewRayTracerFactory;
+  public void addPreviewRenderer(String name, Renderer renderer) {
+    if (InternalRenderManager.previewRenderers.containsKey(name))
+      Log.error("Preview renderer already exists (do you have the same plugin loaded twice?): " + name);
+    InternalRenderManager.previewRenderers.computeIfAbsent(name, n -> renderer);
   }
 
   @PluginApi
-  public RayTracerFactory getPreviewRayTracerFactory() {
-    return previewRayTracerFactory;
+  public void addRenderer(String name, Renderer renderer) {
+    if (InternalRenderManager.renderers.containsKey(name))
+      Log.error("Renderer already exists (do you have the same plugin loaded twice?): " + name);
+    InternalRenderManager.renderers.computeIfAbsent(name, n -> renderer);
   }
 
   @PluginApi
-  public void setRayTracerFactory(RayTracerFactory rayTracerFactory) {
-    this.rayTracerFactory = rayTracerFactory;
+  @Deprecated
+  public void setPreviewRayTracerFactory(RayTracerFactory previewRayTracerFactory) {
+    if (InternalRenderManager.previewRenderers.containsKey("Plugin Preview Renderer"))
+      Log.error("Attempted to register 2 plugin preview renderers.");
```

Comment:
Instead of logging this error I'd just log a warning (on all these methods) that `setPreviewRayTracerFactory` is deprecated and the plugin should be updated.

Reply:
New behavior:
* INFO level log saying `setPreviewRayTracerFactory` is deprecated
* WARN level log warning 2+ plugin preview renderers are installed
* Last loaded plugin wins